src/EFCore.MySql/Scaffolding/Internal/MySqlDatabaseModelFactory.cs

```diff
@@ -509,6 +509,12 @@ private void GetConstraints(DbConnection connection, IReadOnlyList<DatabaseTable
                         {
                             var referencedTableName = reader.GetString(2);
                             var referencedTable = tables.FirstOrDefault(t => t.Name == referencedTableName);
+                            if (referencedTable == null)
+                            {
+                                // if the casing of the constraint doesn't match the casing of the table name, try and find a matching table with a
+                                // different case, but ensure that there's only one possible match or we can't be certain which is the right table.
```

Comment:
Please change the comment to something that describes the issue we handle here. For example:

```c#
// On operation systems with insensitive file name handling, the saved reference table name might have a
// different casing than the actual table name. (#1017)
```